packages/lit-auth-client/src/lib/providers/WebAuthnProvider.ts

```diff
@@ -0,0 +1,121 @@
+import { AuthMethod, BaseProviderOptions, IRelay } from '@lit-protocol/types';
+import { AuthMethodType } from '@lit-protocol/constants';
+import { ethers } from 'ethers';
+import {
+  startRegistration,
+  startAuthentication,
+} from '@simplewebauthn/browser';
+import {
+  PublicKeyCredentialCreationOptionsJSON,
+  UserVerificationRequirement,
+} from '@simplewebauthn/typescript-types';
+import base64url from 'base64url';
+import { getRPIdFromOrigin } from '../utils';
+import { BaseProvider } from './BaseProvider';
+
+export default class WebAuthnProvider extends BaseProvider {
+  constructor(options: BaseProviderOptions) {
+    super(options);
+  }
+
+  /**
+   * Generate registration options for the browser to pass to a supported authenticator
+   *
+   * @param {string} username - Username to register credential with
+   *
+   * @returns {Promise<PublicKeyCredentialCreationOptionsJSON>} - Options to pass to the authenticator
+   */
+  public async register(
+    username?: string
+  ): Promise<PublicKeyCredentialCreationOptionsJSON> {
+    return await this.relay.generateRegistrationOptions(username);
+  }
+
+  /**
+   * Mint PKP with verified registration data
+   *
+   * @param {PublicKeyCredentialCreationOptionsJSON} options - Registration options to pass to the authenticator
+   *
+   * @returns {Promise<string>} - Mint transaction hash
+   */
+  public async verifyAndMintPKPThroughRelayer(
+    options: PublicKeyCredentialCreationOptionsJSON
+  ): Promise<string> {
+    // Submit registration options to the authenticator
+    const attResp = await startRegistration(options);
+
+    // Send the credential to the relying party for verification
+    const mintRes = await this.relay.mintPKP(
+      AuthMethodType.WebAuthn,
+      JSON.stringify({ credential: attResp })
+    );
+    if (!mintRes || !mintRes.requestId) {
+      throw new Error('Missing mint response or request ID from relay server');
+    }
+    // If the credential was verified and registration successful, minting has kicked off
+    return mintRes.requestId;
+  }
+
+  /**
+   * @override
+   * This method is not applicable for WebAuthnProvider and should not be used.
+   * Use verifyAndMintPKPThroughRelayer instead to mint a PKP for a WebAuthn credential.
+   *
+   * @throws {Error} - Throws an error when called for WebAuthnProvider.
+   */
+  public override async mintPKPThroughRelayer(
+    authMethod: AuthMethod
+  ): Promise<string> {
+    throw new Error(
+      'Use verifyAndMintPKPThroughRelayer for WebAuthnProvider instead.'
+    );
+  }
+
+  /**
+   * Authenticate with WebAuthn
+   *
+   * @param {string} [rpcUrl] - RPC URL to use for getting latest block hash
+   *
+   * @returns {Promise<AuthMethod>} - Auth method object containing WebAuthn auth data
+   */
+  public async authenticate(): Promise<AuthMethod> {
```

Comment:
should we add this method signature to `baseProvider`? might be useful for other auth methods in the future.

Reply:
@joshLong145 - added an abstract method `authenticate` to `BaseProvider`, ty!